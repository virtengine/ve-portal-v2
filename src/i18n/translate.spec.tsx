import { mount } from 'enzyme';

import { formatJsx, formatJsxTemplate } from './translate';

describe('formatJsxTemplate', () => {
  it('allows to use curly braces syntax to interpolate JSX component', () => {
    const supportEmail = 'admin@example.com';
    expect(
      mount(
        formatJsxTemplate(
          'Please send an email to {supportEmail}. Thank you!',
          {
            supportEmail: <a href={`mailto:${supportEmail}`}>{supportEmail}</a>,
          },
        ),
      ),
    ).toMatchSnapshot();
  });
});

describe('formatJsx', () => {
  it('allows to use angular braces syntax to interpolate JSX component', () => {
    expect(
      mount(
        formatJsx(
          'By submitting the form you are agreeing to the <Link>Terms of Service</Link>.',
          {
            Link: (s) => <a href="example.com">{s}</a>,
          },
        ),
      ),
    ).toMatchSnapshot();
  });
});
