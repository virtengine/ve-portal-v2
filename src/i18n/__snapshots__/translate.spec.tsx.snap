// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatJsx allows to use angular braces syntax to interpolate JSX component 1`] = `
Array [
  "By submitting the form you are agreeing to the ",
  <a
    href="example.com"
  >
    Terms of Service
  </a>,
  ".",
]
`;

exports[`formatJsxTemplate allows to use curly braces syntax to interpolate JSX component 1`] = `
Array [
  "Please send an email to ",
  <a
    href="mailto:admin@example.com"
  >
    admin@example.com
  </a>,
  ". Thank you!",
]
`;
